### DISCUSSION TITLE
How to obtain IDL command line behaviour in IDL notebooks.

### DISCUSSION THREAD

**Discussion by AvijeetPrasad** [interactive-data-language/vscode-idl#30] [Q&A]
When using IDL command line, we can easily print a variable value directly by typing the variable name as shown below
```
IDL> x = indgen(3)
IDL> x
       0       1       2
```
However, in the notebook, we need to print the variable and not type it directly as shown below
```
x = indgen(3)
print, x
```
I was wondering if there is a way to avoid typing `print` every time and use some kind of "magic" command, say `%` or `!` to mimic the command line behaviour directly.

**Comment by znorman-harris** [comment]
Sorry for missing this! It's on our list to look at doing. The challenge is that the IDL Console has a special way to determine if a command should do implied print or not.

All notebook cells run as main level programs, so we will have to add in logic for this.

I'll take a look at it here in the next couple of weeks and see what it would take to implement.

**Comment by znorman-harris** [comment]
FYI just published a release of this. I updated our example "hello world" notebook that comes with the extension, but in general, it should match the IDL Console.

Here's some examples:

![Screenshot 2024-05-15 154219](https://github.com/interactive-data-language/vscode-idl/assets/31664668/b952c928-8dde-4833-bb13-875d305687c8)
